client/image_remove.go

```diff
@@ -2,14 +2,25 @@ package client
 
 import (
 	"encoding/json"
+	"errors"
 	"net/url"
 
+	"strings"
+
 	"github.com/docker/docker/api/types"
 	"golang.org/x/net/context"
 )
 
 // ImageRemove removes an image from the docker host.
 func (cli *Client) ImageRemove(ctx context.Context, imageID string, options types.ImageRemoveOptions) ([]types.ImageDelete, error) {
+	if strings.TrimSpace(imageID) == "" {
+		return nil, errors.New("image name cannot be blank")
+	}
+
+	if _, err := parseNamed(imageID); err != nil {
+		return nil, err
+	}
```

Comment:
I'm wondering if we should do this validation client-side. Validation is already performed by the daemon (i.e., without these checks;

```
$ docker image rm ""
Error response from daemon: image name cannot be blank

$ docker image rm foo::::bar
Error response from daemon: Error parsing reference: "foo::::bar" is not a valid repository/tag: invalid reference format
```

And a "minimal" client-side check is already done by [`cli.RequiresMinArgs(1)`](https://github.com/docker/docker/blob/b993609d5ad4590da72e217bb43786b74aabc183/cli/command/image/remove.go#L27)

```
docker image rm
"docker image rm" requires at least 1 argument(s).
See 'docker image rm --help'.

Usage:  docker image rm [OPTIONS] IMAGE [IMAGE...]

Remove one or more images
```

Starting with docker 1.13 we support a client talking to a different version of the daemon (both _older_ and _newer_), by moving this validation to the client, there is a possibility (although "slim"), that the client will invalidate values that are actually _valid_ for the daemon.